src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -479,14 +521,19 @@ private void BeginCondenseTransition()
             if (this.IsExpanded == false)
                 throw new InvalidOperationException();
 
-            RefreshCondensedDisplay();
+            //To prevent another transition from being started
+            currentState = State.InTransition;
 
-            this.smallContentGrid.Visibility = System.Windows.Visibility.Visible;
-            SetCurrentStateAndNotify(State.InTransition);
+            RefreshCondensedDisplay(() =>
+                {
+                    this.smallContentGrid.Visibility = System.Windows.Visibility.Visible;
+                    SetCurrentStateAndNotify(State.InTransition);
```

Comment:
If we can use `SetCurrentStateAndNotify` above, then this line should be removed.